modules/launcher/src/main/java/org/mule/runtime/module/launcher/application/DefaultMuleApplication.java

```diff
@@ -172,24 +174,25 @@ public void init()
 
         try
         {
-            ConfigurationBuilder cfgBuilder = domainRepository.getDomain(descriptor.getDomain()).createApplicationConfigurationBuilder(this);
-            if (!cfgBuilder.isConfigured())
+            ArtifactMuleContextBuilder artifactBuilder = new ArtifactMuleContextBuilder()
+                    .setArtifactProperties(descriptor.getAppProperties())
+                    .setArtifactType(APP)
```

Comment:
would it make sense to have a specific builder class for each artifact type? I understand that right now it might be quite overkill, but my concern is that in the future we want to add something that only applies for one of the possible types and then the API becomes ugly